src/Blocks/Expandable.purs

```diff
@@ -71,29 +94,44 @@ contentClasses status = contentSharedClasses <>
       ]
   )
 
+type HTMLexpandable = Interactive ( expanded :: Status )
+
+status :: ∀ r i. Status -> HP.IProp ( expanded :: Status | r ) i
+status = HP.prop (PropName "expanded")
+
+extractStatus
+  :: ∀ r i
+   . Array (HH.IProp ( expanded :: Status | r) i)
+  -> Tuple Status (Array (HH.IProp r i))
+extractStatus =
+  foldr f (Tuple Expanded [])
+  where
+    f (HP.IProp (Property "expanded" expanded)) =
+      lmap (const $ coerceExpanded expanded)
+    f iprop = rmap $ (flip snoc) $ coerceR iprop
+
+    coerceExpanded :: PropValue -> Status
+    coerceExpanded = fromMaybe Expanded <<< readStatus <<< unsafeCoerce
+
+    coerceR :: HH.IProp ( expanded :: Status | r ) i -> HH.IProp r i
+    coerceR = unsafeCoerce
+
 heading
   :: ∀ p i
-   . Status
-  -> Array (HH.IProp HTMLheader i)
+   . Array (HH.IProp HTMLexpandable i)
```

Comment:
This is a nice pattern. @foresttoney from your talk about Ocelot and configuration are there other places you think this should be applied we could create as tasks?

Reply:
I'm all for applying this approach to other blocks, though maybe after we figure out this performance issue with our blocks